api/hypershift/v1alpha1/hostedcluster_types.go

```diff
@@ -1574,6 +1574,12 @@ type AzurePlatformSpec struct {
 	SubnetID          string `json:"subnetID"`
 	SubscriptionID    string `json:"subscriptionID"`
 	SecurityGroupID   string `json:"securityGroupID"`
+
+	// MultiArch specifies whether the Hosted Cluster will be expected to support NodePools with different
+	// CPU architectures, i.e., supporting arm64 NodePools and supporting amd64 NodePools on the same Hosted Cluster.
+	// +kubebuilder:default=false
+	// +optional
+	MultiArch bool `json:"multiArch"`
```

Comment:
At any point, do you need to know which architectures the cluster should be supporting? Eg will you need to know if it's arm and amd vs arm and ppcle?

I wonder if this should be a list, where you can detect the available architectures and handle the multi-architecture case when the list has a length greater than 1.

If there's no benefit to understanding specific architectures, then perhaps an enum of `payloadArchitecture: Single | Multi` might be a good start?

Reply:
The architectures will be amd64, arm64. I don't see ppcle or s390x coming to Azure.

I would lean more towards "there's no benefit to understanding specific architectures".

Reply:
I would probably go down the route of that enum then, in the future if we did need to know the difference, we could expand with another value `Specific` and then have the additional field for users to specify which specific architectures they want